CMakeLists.txt

```diff
@@ -1,22 +1,65 @@
+cmake_minimum_required(VERSION 3.22)
 PROJECT( twbt )
+set( TWBT_VER "6.xx" )
 
-SET( PROJECT_SRCS twbt.cpp )
-SET( TWBT_VER "5.xx" )
+SET(CMAKE_VERBOSE_MAKEFILE ON)
+if(DEFINED dfhack_SOURCE_DIR)
+    set(DFHACK ${dfhack_SOURCE_DIR})
+elseif(EXISTS ../dfhack) # sitting next to each other top level
+    set(DFHACK ../dfhack)
+elseif(EXISTS ../../dfhack) # twbt is inside a dfhack directory, probably dfhack/plugins
+    set(DFHACK ../../dfhack)
+elseif(EXISTS ../../../dfhack) # twbt is inside a directory in dfhack/plugins (probably)
+    set(DFHACK ../../../dfhack)
+else()
+    message("couldn't find a dfhack directory nearby, modify the cmake and put your dfhack path")
+    SET(DFHACK /your/path/dfhack)
```

Comment:
If you're keeping this, I would suggest using a [cached variable](https://cmake.org/cmake/help/latest/command/set.html#set-cache-entry) so that it can be edited by the user in CMakeCache.txt, which is not tracked in source control. DFHack has several examples of this.

The message probably ought to be a FATAL_ERROR as well, so that CMake doesn't proceed with a known-invalid path.

Reply:
good point